ha/core/event_manager/event_manager.py

```diff
@@ -333,19 +334,37 @@ def get_events(self, component : str) -> list:
         """
         pass
 
-    def publish(self, component:str, event: RecoveryActionEvent) -> None:
+    def publish(self, event: RecoveryActionEvent) -> None:
         """
         Publish event.
         Args:
-            component (str): Component name.
             event (RecoveryActionEvent): Action event.
         """
-        pass
+        try:
+            component_list = []
+            # Run through list of components subscribed for this event and send event to each of them
+            component_list_key = f'{const.EVENT_KEY}/{event.resource_type}/{event.event_type}'
+            if self._confstore.key_exists(component_list_key):
```

Comment:
can skip this check.
 _confstore.get will return "None" if key is not found and that "None" can be used to skip the publish call